keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
-+ _BucketAccessClass (BAC)_ - a cluster-scoped custom resource containing fields defining the provisioner and a ConfigMap reference where policy is defined.
-+ _BucketAccessRequest (BAR)_ - a user-namespaced custom resource representing a request for access to an existing bucket.
-+ _BucketClass (BC)_ - a cluster-scoped custom resource containing fields defining the provisioner and an immutable parameter set for creating new buckets.
-+ _BucketRequest (BR)_ - a user-namespaced custom resource representing a request for a new backend bucket, or access to an existing bucket.
-+ _COSI_ - Container _Object_ Store Interface, modeled after CSI.
-+ _cosi-node-adapter_ - a pod per node which receives Kubelet gRPC _NodeStageVolume_ and _NodeUnstageVolume_ requests, ensures the target bucket has been provisioned, and notifies the kubelet when the pod can be run.
-+ _driver_ - a container (usually paired with a sidecar container) that is responsible for communicating with the underlying object store to create, delete, grant access to, and revoke access from buckets. Drivers talk gRPC and need no knowledge of Kubernetes. Drivers are typically written by storage vendors, and should not be given any access outside of their namespace.
-+ _driverless_ - a use-case where no driver exists to support the backend object store. Some COSI automation may still be provided, but backend operations such as creating a bucket or minting credentials will not occur.
-+ _greenfield bucket_ - a new backend bucket created by COSI.
-+ _green-to-brownfield_ - a use-case where COSI creates a new bucket on behalf of a user, and then supports ways for other users in the cluster to share this bucket.
-+ _provisioner_ - a generic term meant to describe the combination of a sidecar and driver. "Provisioning" a bucket can mean creating a new bucket or granting access to an existing bucket.
-+ _sidecar_ - a Kubernetes-aware container (usually paired with a driver) that is responsible for fullfilling COSI requests with the driver via gRPC calls. The sidecar's access can be restricted to the namespace where it resides, which is expected to be the same namespace as the provisioner. The COSI sidecar is provided by the Kubernetes community.
-+ _static provisioning_ - custom resource creation is done by the admin rather than via COSI automation. This may also include _driverless_ but they are independent.
+#### Functionality
 
-# Proposal
++ Support automated **Bucket Creation**
++ Support automated **Bucket Deletion**
++ Support automated **Access Credential Generation**
++ Support automated **Access Credential Revokation**
++ Support automated **Bucket Provisioning** for workloads (enabling pods to access buckets)
++ Support **Bucket Reuse** 
 
-## APIs
+#### System Properties
 
-### Storage APIs
++ Support **workload portability** across clusters
++ Achieve the above goals in a **vendor neutral** manner
++ Standardize mechanism for third-party vendors to integrate easily with COSI
```

Comment:
should idempotency be mentioned here and the expectation that drivers are also idempotent?

Reply:
I'm trying to reserve that level of granularity in the individual sections rather than at the high level.